Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -245,6 +245,61 @@ void WebExtension::populateWebAccessibleResourcesIfNeeded()
         parseWebAccessibleResourcesVersion2();
 }
 
+URL WebExtension::resourceFileURLForPath(const String& originalPath)
+{
+    ASSERT(originalPath);
+
+    String path = originalPath;
+
+    if (path.startsWith('/'))
+        path = path.substring(1);
+
+    if (!path.length() || m_resourceBaseURL.isNull())
+        return { };
+
+    auto resourceURL = URL(m_resourceBaseURL, path).fileSystemPath();
+
+    // Don't allow escaping the base URL with "../".
+    if (!resourceURL.startsWith(m_resourceBaseURL.string())) {
+        RELEASE_LOG_ERROR(Extensions, "Resource URL path escape attempt: %s", resourceURL.utf8().data());
+        return { };
+    }
+
+    return URL(resourceURL);
```

Comment:
This will not be a `file` URL. I suspect this will fail to parse, since `resourceURL` will just be a `/foo/bar/path` string.